packages/components/src/select-control/index.js

```diff
@@ -92,21 +93,29 @@ export class SelectControl extends Component {
 	}
 
 	selectOption( option ) {
-		const { multiple, onChange, selected } = this.props;
-		const { query } = this.state;
+		const { multiple, selected } = this.props;
 		const newSelected = multiple ? [ ...selected, option ] : [ option ];
 
 		this.reset( newSelected );
 
+		const oldSelected = Array.isArray( selected )
+			? selected
+			: [ { key: selected } ];
+		const isSelected = findIndex( oldSelected, { key: option.key } );
+		if ( isSelected === -1 ) {
+			this.setNewValue( newSelected );
+		}
+	}
+
+	setNewValue( newValue ) {
```

Comment:
Have all value changes go through the `setNewValue` method that checks whether to set it as an array or string.